packages/ui-icons-lucide/src/wrapLucideIcon/index.tsx

```diff
@@ -143,25 +92,72 @@ export function wrapLucideIcon(Icon: LucideIcon): LucideIcon {
       accessibilityProps['role'] = 'presentation'
     }
 
+    const gradientId = useId()
+
+    // AI Gradient Implementation:
+    // SVG gradients must be defined BEFORE they're referenced. Since Lucide renders
+    // icon paths before any children we pass, we inject the gradient in a separate
+    // hidden SVG element that comes BEFORE the icon in the DOM. We use
+    // gradientUnits="userSpaceOnUse" with coordinates (0,0) to (24,24) matching
+    // Lucide's viewBox, ensuring one gradient spans the entire icon space rather
+    // than scaling separately for each shape (which causes small elements to lose
+    // gradient visibility). The icon then references this gradient via stroke="url(#id)".
+    if (styles.gradientColors) {
+      // Use viewBox coordinates for gradient (Lucide icons use 0 0 24 24 viewBox)
+      const gradientSize = 24
+
+      return (
+        <span css={styles.lucideIcon}>
+          {/* Hidden SVG to define the gradient - must come before the icon uses it */}
+          <svg width={0} height={0} style={{ position: 'absolute' }}>
+            <defs>
+              <linearGradient
+                id={gradientId}
+                x1="0"
+                y1="0"
+                x2={gradientSize}
+                y2={gradientSize}
+                gradientUnits="userSpaceOnUse"
+              >
+                <stop offset="0%" stopColor={styles.gradientColors.top} />
+                <stop offset="100%" stopColor={styles.gradientColors.bottom} />
+              </linearGradient>
+            </defs>
+          </svg>
+          <Icon
+            {...passthroughProps(rest)}
+            name={Icon.displayName}
+            ref={handleElementRef}
+            size={styles.numericSize}
+            color={`url(#${gradientId})`}
+            strokeWidth={styles.numericStrokeWidth}
+            absoluteStrokeWidth={absoluteStrokeWidth}
+            {...accessibilityProps}
+          />
+        </span>
+      )
+    }
+
+    // Normal rendering (non-gradient)
     return (
-      <span css={styles?.lucideIcon} className={className} style={style}>
+      <span css={styles.lucideIcon}>
         <Icon
+          {...passthroughProps(rest)}
           name={Icon.displayName}
           ref={handleElementRef}
-          size={numericSize}
-          color={customColor}
-          strokeWidth={numericStrokeWidth}
+          size={styles.numericSize}
+          color={styles.customColor}
+          strokeWidth={styles.numericStrokeWidth}
           absoluteStrokeWidth={absoluteStrokeWidth}
           {...accessibilityProps}
-          {...passthroughProps(rest)}
         />
       </span>
     )
   }
 
   WrappedIcon.displayName = `wrapLucideIcon(${Icon.displayName || Icon.name})`
```

Comment:
Why is this displayname different from the one in line 83? It should be the same for consistency and theme override's sake